pandas/core/arrays/_ranges.py

```diff
@@ -14,14 +14,15 @@
     Timestamp,
     iNaT,
 )
+from pandas._typing import npt
 
 
 def generate_regular_range(
-    start: Timestamp | Timedelta,
-    end: Timestamp | Timedelta,
-    periods: int,
+    start: Timestamp | Timedelta | None,
+    end: Timestamp | Timedelta | None,
+    periods: int | None,
```

Comment:
Nit: Mind updating the docstring below for this argument that periods can be None?

Reply:
green and updated the doc-string